software/teachee-desktop/src/app.rs

```diff
@@ -82,29 +112,143 @@ impl eframe::App for App {
             channel1_offset,
             channel2,
             channel2_offset,
+            h_offset,
+            h_scale,
+            v_offset,
+            v_scale,
+            saved_v_offset,
+            saved_v_scale,
+            channel1_v_offset,
+            channel1_v_scale,
+            channel2_v_offset,
+            channel2_v_scale,
+            channel1_on,
+            channel2_on,
+            trigger_button_text,
             ..
         } = self;
 
-        SidePanel::right("controls")
-            .resizable(false)
-            .show(ctx, |ui| {
-                ScrollArea::vertical()
-                    .auto_shrink([false, false])
-                    .show(ui, |ui| {
-                        channel_control(ui, "Channel 1", channel1, channel1_offset);
-                        channel_control(ui, "Channel 2", channel2, channel2_offset);
-                    });
-            });
-
         TopBottomPanel::top("top").show(ctx, |ui| {
             ui.horizontal_wrapped(|ui| {
                 ui.visuals_mut().button_frame = false;
                 widgets::global_dark_light_mode_switch(ui);
                 ui.separator();
-                ui.label(if *flag { "Flag: 1" } else { "Flag: 0" });
+                ui.menu_button("File", |ui| {
+                    if ui.button("Export to CSV").clicked() {
+                        // TODO: Export to CSV
```

Comment:
Check out https://doc.rust-lang.org/std/macro.todo.html. Not important in this case, but it's handy since it makes unfinished code typecheck. Though it will crash the program if the button is clicked.